core/data-layer/src/data_layer.rs

```diff
@@ -696,6 +731,121 @@ impl DataLayer {
+         *  source metrics metadata - LinkedList<(metric_id, ULID, data size(source metrics + source metrics metadata)> - list order by ULID ASC */
+        let mut source_metrics = self.source_metrics_data.source_metrics.write().await;
+        let mut source_metrics_metadata = self
+            .source_metrics_data
+            .source_metrics_metadata
+            .write()
+            .await;
+        let mut source_metrics_size = self.source_metrics_data.source_metrics_size.write().await;
+        let metric_buffer_size_kb = self.source_metrics_data.metric_buffer_size_kb;
+
+        let now_ulid = Ulid::new().to_string();
+        let source_metric_insert_data = SourceMetrics {
+            json_value: json_value.to_string(),
+        };
+        // source_metric_insert_data_size = source metrics + source metrics metadata
+        let source_metric_insert_data_size = source_metric_insert_data.get_heap_size()
+            + (metric_id.to_string().get_heap_size() * 2)
+            + (now_ulid.get_heap_size() * 2);
+
+        let mut total_source_metrics_size = *source_metrics_size + source_metric_insert_data_size;
+
+        // get remove target data
+        let mut remove_target_data: Vec<(String, String, usize)> = Vec::new();
+        let mut subtract_total_size = total_source_metrics_size;
+        loop {
+            // check size :: buffersize - total size < 0 (None) => remove target data
+            if metric_buffer_size_kb
+                .checked_sub(subtract_total_size as u64)
+                .is_none()
+            {
+                let Some(front_source_metrics_metadata) = source_metrics_metadata.pop_front() else {
+                        break;
+                    };
+                subtract_total_size -= front_source_metrics_metadata.2; // oldest metadata size
+                remove_target_data.append(&mut vec![front_source_metrics_metadata]);
+                continue;
+            }
+            break;
+        }
+
+        // remove target data
+        remove_target_data
+            .iter()
+            .for_each(|(metric_id, ulid, size)| {
+                let Some(metric_id_data) = source_metrics.get_mut(metric_id) else {
+                    return;
+                };
+                // remove source metrics - btreemap ulid
+                metric_id_data.remove(ulid);
+                // (if btreemap is empty) remove source metrics - hashmap metric_id
+                if metric_id_data.is_empty() {
+                    source_metrics.remove(metric_id);
+                }
+                total_source_metrics_size -= size;
+            });
+
+        // add source metrics
+        match source_metrics.get_mut(metric_id) {
+            Some(source_metrics_map) => {
+                source_metrics_map.insert(now_ulid.clone(), source_metric_insert_data.clone());
+            }
+            None => {
+                let mut metric_id_map = BTreeMap::new();
```

Comment:
```suggestion
    let mut source_metrics_map = BTreeMap::new();
```
In order to match variable names as shown above